### PR TITLE
fixing some warnings.

### PR DESCRIPTION
**Pull request by GitPaean**
Silencing some warnings from `mpi.h`, and a few other warning silencing.

### CODE REVIEW

File: dune/grid/common/ZoltanGraphFunctions.cpp [hunk 1]
```diff
@@ -167,7 +167,7 @@ void getNullEdgeList(void *cpGridPointer, int sizeGID, int sizeLID,
     (void) cpGridPointer; (void) sizeGID; (void) sizeLID; (void) numCells;
     (void) globalID; (void) localID; (void) numEdges; (void) nborGID;
     (void) nborProc; (void) wgtDim; (void) ewgts;
-    err = ZOLTAN_OK;
```
Comment (andlaus) [inline dune/grid/common/ZoltanGraphFunctions.cpp]: hu? how could that code have ever worked?

Comment (bska) [inline dune/grid/common/ZoltanGraphFunctions.cpp]: One possible mechanism is if `ZOLTAN_OK` is `#define`d as zero (can't check right now) in some header.  In that case the statement
```C++
err = ZOLTAN_OK;
```
makes `err` into a null pointer within the function body and there's no change visible in the caller.  If, moreover, the caller already initialises `*err` as `ZOLTAN_OK` before calling into the call-back then the call-back is a big no-op.

Comment (bska) [inline dune/grid/common/ZoltanGraphFunctions.cpp]: > so does that boil down to: it worked because of unlucky coincidences?

Quite possibly, yes.

**Comment by bska** [comment]
jenkins build this please

File: dune/grid/common/ZoltanGraphFunctions.cpp (hunk 1 above)
Comment (blattms) [inline dune/grid/common/ZoltanGraphFunctions.cpp]: Sorry, did not look further down. Fix is already there.

**Comment by atgeirr** [comment]
Good! (Warnings rule!)

**Pull request merged**